src/structs.rs

```diff
@@ -435,14 +436,64 @@ impl Serializable for IndexedValue {
 #[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
 pub struct Uid<const LENGTH: usize>([u8; LENGTH]);
 
+impl<const UID_LENGTH: usize> Display for Uid<UID_LENGTH> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", STANDARD.encode(self.0))
+    }
+}
+
 impl_byte_array!(Uid);
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Uids<const LENGTH: usize>(pub HashSet<Uid<LENGTH>>);
+
+impl<const LENGTH: usize> Deref for Uids<LENGTH> {
+    type Target = HashSet<Uid<LENGTH>>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl<const UID_LENGTH: usize> IntoIterator for Uids<UID_LENGTH> {
+    type IntoIter = <<Self as Deref>::Target as IntoIterator>::IntoIter;
+    type Item = Uid<UID_LENGTH>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.0.into_iter()
+    }
+}
+
+impl<const UID_LENGTH: usize> Display for Uids<UID_LENGTH> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut output = String::new();
+        for uid in self.clone() {
+            output = format!("uid: {}\n{output}", STANDARD.encode(uid));
```

Comment:
Is this really what you want ? I mean it seems to me that only the last UID is displayed, but maybe there is something I didn't get.

Reply:
it also takes the `output` variable for each element of the list

Reply:
Ah okay, I got it, you include `output` in the `format!()`... why don't you simply extend the output with the new `format` ?

Reply:
done in https://github.com/Cosmian/findex/pull/65/commits/9b7dde9771647d197ad4474ccdec55e0be60bd8a